### ISSUE TITLE
Support CompletableFuture<T>

### ISSUE THREAD

**Issue by d-gs** [zeroturnaround/zt-exec#76]
I would like to use `CompletableFuture<ProcessResult>` compose functionality instead of a pure `Future` to be able to chain process as in a pipeline waiting one for the next. Looking a into your code (not soo deeply) I found that this might be simple by modifying the following method to use [`CompletableFuture#supplyAsync(Supplier, Executor)`]( https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#supplyAsync-java.util.function.Supplier-java.util.concurrent.Executor-) instead of `ExecutorService#submit(Callable<T>)`:

https://github.com/zeroturnaround/zt-exec/blob/08dda48d02b76f17b9fd2f1f55cf2b008ffc242c/src/main/java/org/zeroturnaround/exec/ProcessExecutor.java#L1147

Changing the signature of the methods and variables for `CompletableFuture` should not have any impact in clients of the API (but maybe on custom implementations). Another putative side-effect is that the shutdown of the `ExecutorService` might fail if chaining in an asynchronous way the produced `CompletableFuture`.

Could it be possible to implement such a solution? Thanks!

**Comment by nemecec** [comment]
Would you be willing to send a pull request?
(Sorry for the long delay, I'm taking over maintenance of this project so communication should be faster from now on)

**Comment by haiderkagalwala** [comment]
I would like to contribute to some of these issues. Working on a similar project so I have the background. Is the project still active and acceptings PRs?

**Comment by nemecec** [comment]
Project is semi-active. Should find some time to clean it up and publish a release.

**Comment by MastaP** [comment]
@nemecec delegate to Claude ;-)

**Comment by nemecec** [comment]
> [@nemecec](https://github.com/nemecec) delegate to Claude ;-)

Yes, that is the plan. But even that requires some attention from me.
